File: docs/snippets/cas/tokenizer_tokenize.py
from typing import List
from mathy.core.tokenizer import Tokenizer, Token

text = "4x + 2x^3 * 7x"
tokenizer = Tokenizer()
tokens: List[Token] = tokenizer.tokenize(text)

for token in tokens:
    print(f"type: {token.type}, value: {token.value}")

File: docs/snippets/cas/tokenizer_tokenize.ts
import { Tokenizer, Token } from "mathy/core/tokenizer"

const text = "4x + 2x^3 * 7x"
const tokenizer = new Tokenizer()
const tokens: Token[] = tokenizer.tokenize(text)
for (token in tokens) {
  console.log(`type: ${token.type}, value: ${token.value}`)
}
